src/main/lrsql/util/auth.clj

```diff
@@ -14,7 +14,8 @@
   {"all"                  :scope/all
    "all/read"             :scope/all.read
    "statements/read"      :scope/statements.read
-   "statements/write"     :scope/statements.write})
+   "statements/write"     :scope/statements.write
+   "admin"                :scope/admin})
```

Comment:
I'm calling this out as a possible smell, I added the `admin` scope (expressing the ability to do admin things) to the same scopes used for xAPI. This was more out of a desire to dedupe code than anything else, but it might not be great.

Reply:
These auth levels are strictly for statement insert/query and other LRS ops. I don't think it makes sense to add an "admin" scope since in lrsql _every_ user is effectively an admin, i.e. they can create or delete both accounts and API key pairs.

Reply:
> These auth levels are strictly for statement insert/query and other LRS ops.

Agreed, it doesn't belong there

> I don't think it makes sense to add an "admin" scope since in lrsql every user is effectively an admin, i.e. they can create or delete both accounts and API key pairs.

I don't agree with this. If we don't require *any* scope for admin actions, there is nothing to differentiate a user who for instance has the `statements/write` scope from one that can create/modify accounts and credentials. Proper OIDC support for admin stuff needs a scope, I just don't think it should be here per se.

Reply:
Or put another way, it makes the xAPI scopes meaningless if any identity on the IDP can just make some new creds with the `all` scope just by virtue of being in the system

Reply:
Consider an organization with an existing IDP. They might have 1000 users in the IDP. Within that, they want 500 of those users to be able to post statements and only 10 to be able to administer it. Scopes!

Reply:
I see - before we had a clear distinction of who's an admin and who's just a LRS statement reader/writer based on what they use to authenticate (API key pairs in the latter case). If you're using OIDC for both, that distinction breaks down, so the old auth model of having just scopes for statements becomes obsolete.

Reply:
Yep, that's pretty much it. You have to have some way to distinguish them, it's implicit in the local/vanilla auth (and that's fine), but you need to differentiate if you're federating with a larger system

Reply:
In that case there's definitely going to be some surgery done to the `authorize-action` function. I wonder if any additional changes should be made to the `lrs` library to make it easier to work with OIDC (I'm certain it's not required though).

Reply:
Well again, the LRS lib knows nothing about administration, and I don't know that it should. It delegates this stuff downstream because it is hard to generalize. See https://github.com/yetanalytics/lrsql/pull/207/commits/29126d851ccf3109e572963561bf2b46983650da for an extraction of the admin scope to the OIDC ns